AK/StringView.cpp

```diff
@@ -75,6 +75,34 @@ Vector<StringView> StringView::split_view(const char separator, bool keep_empty)
     return v;
 }
 
+Vector<StringView> StringView::split_view(const StringView& separator, bool keep_empty) const
+{
+    if (is_empty())
+        return {};
+
+    StringView view { *this };
+
+    Vector<StringView> parts;
+    if (separator.is_empty()) {
+        parts.append(view);
+        return parts;
+    }
+
+    auto maybe_separator_index = find_first_of(separator);
+    while (maybe_separator_index.has_value()) {
```

Comment:
Now we can probably make this into a `for` loop, because it really is one:

```c++
for (
    auto maybe_separator_index = find_first_of(separator);
    maybe_separator_index.has_value();
    maybe_separator_index = find_first_of(separator)
)
```

Reply:
I mean...that reads like 'please don't do this to me'

Reply:
No, really, I like the `while` version better.

```c++
    for (
        auto maybe_separator_index = find_first_of(separator);
        maybe_separator_index.has_value();
        maybe_separator_index = view.find_first_of(separator)) {

        auto separator_offset = maybe_separator_index.value();
        auto part_with_separator = view.substring_view(0, separator_offset + separator.length());
        if (keep_empty || separator_offset > 0)
            parts.append(part_with_separator.substring_view(0, separator_offset));
        view = view.substring_view_starting_after_substring(part_with_separator);
    }
```
...meh

Reply:
Feel free not to do this if you don't like it,

but my general logic is to use `while` loops for, you know, doing stuff while a condition is true, and `for` loops for iterating over things. Especially loops where you have actions that can be clearly classified as *moving to the next step* (rather than processing this current item) look like `for` loops to me. This is not a 100% strict rule, of course, and it's not me who writes this code, so I'm completely fine with not doing it.

Reply:
The next-step thing would have us doing this-
```c++
 for (
        auto maybe_separator_index = find_first_of(separator);
        maybe_separator_index.has_value();
        view = view.substring_view_starting_after_substring(part_with_separator),
        maybe_separator_index = view.find_first_of(separator)) {

        auto separator_offset = maybe_separator_index.value();
        auto part_with_separator = view.substring_view(0, separator_offset + separator.length());
        if (keep_empty || separator_offset > 0)
            parts.append(part_with_separator.substring_view(0, separator_offset));
    }
```

yeah no.